pkg/nto/tuned.go

```diff
@@ -0,0 +1,269 @@
+
+	builder := &TunedBuilder{
+		apiClient: apiClient.Client,
+		Definition: &tunedv1.Tuned{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the Tuned is empty")
+
+		builder.errorMsg = "tuned 'name' cannot be empty"
+
+		return builder
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The nsname of the tuned is empty")
+
+		builder.errorMsg = "tuned 'nsname' cannot be empty"
+
+		return builder
+	}
+
+	return builder
+}
+
+// PullTuned pulls existing Tuned from cluster.
+func PullTuned(apiClient *clients.Settings, name, nsname string) (*TunedBuilder, error) {
+	glog.V(100).Infof("Pulling existing Tuned name %s in namespace %s from cluster", name, nsname)
+
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient is empty")
+
+		return nil, fmt.Errorf("tuned 'apiClient' cannot be empty")
+	}
+
+	builder := TunedBuilder{
+		apiClient: apiClient.Client,
+		Definition: &tunedv1.Tuned{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the tuned is empty")
+
+		return nil, fmt.Errorf("tuned 'name' cannot be empty")
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the tuned is empty")
+
+		return nil, fmt.Errorf("tuned 'nsname' cannot be empty")
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("tuned object %s does not exist in namespace %s", name, nsname)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches the defined Tuned from the cluster.
+func (builder *TunedBuilder) Get() (*tunedv1.Tuned, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting Tuned %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	tunedObj := &tunedv1.Tuned{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, tunedObj)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return tunedObj, nil
+}
+
+// Create makes a tuned in the cluster and stores the created object in struct.
+func (builder *TunedBuilder) Create() (*TunedBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the tuned %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// Delete removes tuned from a cluster.
+func (builder *TunedBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the tuned %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
```

Comment:
Please don't return error from this function. If you want to delete object that does't exist just print log, set object to nil and return nil.

```
	if !builder.Exists() {
		glog.V(100).Infof("application %s in namespace %s cannot be deleted because it does not exist",
			builder.Definition.Name, builder.Definition.Namespace)

		builder.Object = nil

		return builder, nil
	}
```